File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.cpp
class Solution {
public:
    int maxFreqSum(string s) {
        string vowels = "aeiou";
        unordered_map<char, int> vowelF, consonantF;

        for (const char& c : s){
            if (vowels.find(c) != string::npos)
                ++vowelF[c];
            else
                ++consonantF[c];
        }

        int maxVowel = 0, maxConsonant = 0;

        for (auto& [_, f] : vowelF) maxVowel = max(maxVowel, f);
        for (auto& [_, f] : consonantF) maxConsonant = max(maxConsonant, f);
        return maxVowel + maxConsonant;
    }
};

File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.py
class Solution:
    def maxFreqSum(self, s: str) -> int:
        vowels_f, consonants_f = dict(), dict()
        vowels = "aeiou"

        for c in s:
            if c in vowels:
                vowels_f[c] = vowels_f.get(c, 0) + 1
            else:
                consonants_f[c] = consonants_f.get(c, 0) + 1
        
        return (max(vowels_f.values()) if len(vowels_f) > 0 else 0) + (max(consonants_f.values()) if len(consonants_f) > 0 else 0)